demos/sample_azure_iot_pnp/sample_azure_iot_pnp.c

```diff
@@ -0,0 +1,1051 @@
+                                                           ucCommandPayloadBuffer,
+                                                           ulCommandPayloadLength ) != eAzureIoTSuccess )
+                {
+                    LogError( ( "Error sending command response" ) );
+                }
+            }
+        }
+        else
+        {
+            LogError( ( "Error generating command payload: 0x%08x", xResult ) );
+
+            if( AzureIoTHubClient_SendCommandResponse( pxHandle, pxMessage, 501,
+                                                       ( const uint8_t * ) sampleazureiotCOMMAND_EMPTY_PAYLOAD,
+                                                       strlen( sampleazureiotCOMMAND_EMPTY_PAYLOAD ) ) != eAzureIoTSuccess )
+            {
+                LogError( ( "Error sending command response" ) );
+            }
+        }
+    }
+    else
+    {
+        /* Not for max min report (not for this device) */
+        LogInfo( ( "Received command is not for this device: %.*s",
+                   pxMessage->usCommandNameLength,
+                   pxMessage->pucCommandName ) );
+
+        if( AzureIoTHubClient_SendCommandResponse( pxHandle, pxMessage, 404,
+                                                   ( const uint8_t * ) sampleazureiotCOMMAND_EMPTY_PAYLOAD,
+                                                   strlen( sampleazureiotCOMMAND_EMPTY_PAYLOAD ) ) != eAzureIoTSuccess )
+        {
+            LogError( ( "Error sending command response" ) );
+        }
+    }
+}
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Properties callback handler
+ */
+static AzureIoTResult_t prvProcessProperties( AzureIoTHubClientPropertiesResponse_t * pxMessage,
+                                              AzureIoTHubClientPropertyType_t xPropertyType,
+                                              double * pxOutTemperature,
+                                              uint32_t * ulOutVersion )
+{
+    AzureIoTResult_t xResult;
+    AzureIoTJSONReader_t xReader;
+    const uint8_t * pucComponentName = NULL;
+    uint32_t ulComponentNameLength = 0;
+
+    *pxOutTemperature = 0.0;
+
+    xResult = AzureIoTJSONReader_Init( &xReader, pxMessage->pvMessagePayload, pxMessage->ulPayloadLength );
+    configASSERT( xResult == eAzureIoTSuccess );
+
+    xResult = AzureIoTHubClientProperties_GetPropertiesVersion( &xAzureIoTHubClient, &xReader, pxMessage->xMessageType, ulOutVersion );
+
+    if( xResult != eAzureIoTSuccess )
+    {
+        LogError( ( "Error getting the property version" ) );
+    }
+    else
+    {
+        /* Reset JSON reader to the beginning */
+        xResult = AzureIoTJSONReader_Init( &xReader, pxMessage->pvMessagePayload, pxMessage->ulPayloadLength );
+        configASSERT( xResult == eAzureIoTSuccess );
+
+        while( ( xResult = AzureIoTHubClientProperties_GetNextComponentProperty( &xAzureIoTHubClient, &xReader,
```

Comment:
What happens if a non-NULL componentName comes back from this?  We should log a message and continue processing otherwise.  Is is, if there was a `sampleazureiotPROPERTY_TARGET_TEMPERATURE_TEXT` but on a component we'd process which we don't want.